Source/Core/IO/Lump.cs

```diff
@@ -54,6 +54,9 @@ public class Lump
 		// Disposing
 		private bool isdisposed = false;
 
+        // bmsq - string to longname index, used when name.length > 8
+        private static Dictionary<string, long> longnameindex = new Dictionary<string, long>();
```

Comment:
I'm not overly happy with this living in the Lump class.  Ideally it should be in the DataManager but initialization order is a problem.   It should also be cleared each time a new map is loaded.  

This works as a proof of concept for now.